src/components/AddItemModal.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import DefaultModal, { Overlay, ButtonField } from './DefaultModal';
+import Logo from '../molecules/Logo';
+import TextInput from './TextInput';
+import CancelButton from './CancelButton';
+import MainButton from './MainActionButton';
+
+const Body = styled.div`
+	height: 20em;
+`;
+
+const Desc = styled.p`
+	color: black;
+	font-size: 14px;
+`;
+
+const BodyLogo = styled.div`
+	grid-area: logo;
+	justify-content: center;
+	align-items: center;
+`;
+
+const InputField = styled.div`
+	grid-area: input;
+	margin: 0 2em;
+	display: grid;
+	grid-template-columns: 1fr 1fr;
+	grid-template-rows: 1fr 1.5fr;
+	grid-template-areas:
+		'name date'
+		'weight grade'
+	color: black;
+	justify-content: center;
+	align-items: center;
+`;
+
+const InputName = styled(TextInput)`
+	grid-area: name;
+`;
+
+const InputDate = styled(TextInput)`
+	grid-area: date;
+`;
+
+const InputWeight = styled(TextInput)`
+	grid-area: weight;
+`;
+
+const InputGrade = styled(TextInput)`
+	grid-area: grade;
+`;
+
+const Cancel = styled(CancelButton)`
+	grid-area: left;
+`;
+
+const Submit = styled(MainButton)`
+	grid-area: right;
+`;
+
+export interface AddItemInterface {
+	onSubmit: (data: AddItemData) => void;
+}
+
+export interface AddItemData {
+	name: string;
+	date: string;
+	weight: string;
+	grade: string;
+}
+
+function AddItemModal({ onSubmit }: AddItemInterface) {
+	const [isOpen, setStatus] = useState('');
+	const [name, setName] = useState('');
+	const [date, setDate] = useState('');
+	const [weight, setWeight] = useState('');
+	const [grade, setGrade] = useState('');
+
+	function close() {
+		setStatus('closed');
+	}
+
+	function handleSubmit({ name, date, weight, grade }: AddItemData) {
+		onSubmit({ name, date, weight, grade });
+		close();
+	}
+
+	return (
+		<Overlay onClick={close} className={isOpen}>
```

Comment:
I am not sure I understand how className={isOpen} ?

Reply:
Since `isOpen` is either `'closed'` or `''` my idea was to set `className` equal to one of those strings, where CSS is able to change the display if `className={'closed'}`